### ISSUE TITLE
Configuration options - production / development / bind port and so on

### ISSUE THREAD

**Issue by ninjabear** [nickel-org/nickel.rs#161]
It would be nice to have the ability to switch configurations in the same way that ruby/rails does.

Basically switching the mode to ```development``` switches off caching and puts out more detailed error messages (detailed 500s for example). 

Switching off caching means that you can work on front-end stuff like templates without recompiling and rerunning your application (the most extreme example being something like [the shotgun gem](https://rubygems.org/gems/shotgun)). And in turn production modes switch on all the caching and clever stuff for production use.

The bind port/ip isn't really a big issue but again, sometimes your production port may differ to whatever one you're messing around with while developing. 

This is a fairly low priority item imo.

**Comment by SimonPersson** [comment]
It'd be nice to store all static files in the binary when building for production too. Simplifies deployment a bit if you don't have to keep the folder structure and so on.

**Comment by ninjabear** [comment]
@SimonPersson agree. Java has a resources/ folder, something like that would be nice (though I appreciate rust doesn't produce a zip)

**Comment by SimonPersson** [comment]
I mean, you can already do this to some extent. Provided you don't have too many static files to serve, this works:
```
router.get("/index.htm", middleware! {
    include_str!("static/index.htm")
});
```

I bet you could also write a macro that creates a route like that for every file in a directory. We just need to fix up the templates.
